packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label
+   */
+  labelId: PropTypes.string.isRequired,
```

Comment:
I think only one of these IDs should be necessary. Is there a reason both `id` and `labelId` are required props?

Reply:
Id is required for button label association and it has been referred in `htmlFor` attribute of label element and labelId is required to specify value for aria-labelledBy attribute on button.

Reply:
You don't need to supply `htmlFor` if the control is rendered inside the `label` element

Reply:
we might need it for linter. I tried removing, it throws lint error `A form label must be associated with a control.eslint(jsx-a11y/label-has-associated-control)`

Reply:
I would look at how we're handling this in our other form components. We are not requiring a `labelId` anywhere else.

Reply:
Yes, adding extra prop for label is not preferable. but I don't find any other way either. I checked in other form controls most of them either have `aria-label` attribute or `text` attribute or content between closing and opening tag. In this case native button is missing `aria-label` because adding aria-label will make the screen readers to read aria-label content as well with default `Switch` role. 
I tried adding text between open and closed tags of button even after hiding content from displaying (display : none) it has some weird effects (displays white box within slider only when we hold space or return key for longer time) in safari. hence added aria-labelled-by